linera-service/src/exporter/state.rs

```diff
@@ -0,0 +1,183 @@
+// Copyright (c) Zefchain Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use std::sync::{
+    atomic::{AtomicU64, Ordering},
+    Arc,
+};
+
+use linera_base::{crypto::CryptoHash, data_types::BlockHeight, identifiers::ChainId};
+use linera_client::config::DestinationId;
+use linera_sdk::{
+    ensure,
+    views::{RootView, View},
+};
+use linera_views::{
+    context::Context, log_view::LogView, map_view::MapView, register_view::RegisterView,
+    views::ClonableView,
+};
+use serde::{Deserialize, Serialize};
+
+use crate::common::{BlockId, ExporterError, LiteBlockId};
+
+/// State of the linera exporter as a view.
+#[derive(Debug, RootView, ClonableView)]
+pub struct BlockExporterStateView<C> {
+    /// The causal state.
+    canonical_state: LogView<C, CryptoHash>,
+    /// The chain status, by chain ID.
```

Comment:
This one is probably unnecessary. We already see it's a map keyed by the chainID. The second line is enough.